M2/Macaulay2/e/eigen.cpp

```diff
@@ -0,0 +1,216 @@
+#include <cstdlib>
+
+#if 0
+//void* my_malloc( std::size_t size );
+//void my_free( void* ptr );
+namespace std {
+void* my_malloc( std::size_t size );
+void my_free( void* ptr );
+}
+using std::my_malloc;
+using std::my_free;
+
+#define malloc my_malloc
+#define free my_free
+#endif
+
+#define mpfr foo
+#include "mpreal.h"
+#include <unsupported/Eigen/MPRealSupport>
+#undef mpfr
+#include "eigen.hpp"
+#include <Eigen/SVD>
+
+#if USING_MPIR 
+#include <mpir.h>
+#include <mpirxx.h>
+#else
+#include <gmp.h>
+#include <gmpxx.h>
+#endif
+#include <mpfr.h>
```

Comment:
does this file need this?

Reply:
yes, but it's included from elsewhere  now